parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -295,11 +273,7 @@ pub mod pallet {
 		// that happens just return error so to pause processing FinalizedHeader until
 		// ExecutionHeader catch up
 		fn cross_check_execution_state() -> DispatchResult {
-			let latest_finalized_state: CompactBeaconState =
-				match Self::finalized_beacon_state(Self::latest_finalized_block_root()) {
-					Some(finalized_beacon_state) => finalized_beacon_state,
-					None => return Err(Error::<T>::NotBootstrapped.into()),
-				};
+			let latest_finalized_state: CompactBeaconState = Self::get_latest_finalized_state()?;
```

Comment:
I think rust's type inference means we don't need to explicitly specify the type.
```suggestion
			let latest_finalized_state = Self::get_latest_finalized_state()?;
```